tt_metal/impl/dispatch/command_queue_interface.hpp

```diff
@@ -558,6 +589,17 @@ class SystemMemoryManager {
         } else {
             cq_interface.issue_fifo_wr_ptr += push_size_16B;
         }
+
+        // Also store this data in hugepages, so if a hang happens we can see what was written by host.
+        chip_id_t mmio_device_id = tt::Cluster::instance().get_associated_mmio_device(this->device_id);
+        uint16_t channel = tt::Cluster::instance().get_assigned_channel_for_device(this->device_id);
+        tt::Cluster::instance().write_sysmem(
+            &cq_interface.issue_fifo_wr_ptr,
+            sizeof(uint32_t),
+            HOST_CQ_ISSUE_WRITE_PTR + get_relative_cq_offset(cq_id, this->cq_size),
+            mmio_device_id,
+            channel
+        );
```

Comment:
Not so worried about similar changes in `completion_queue_wait_front `, but do we know if this impacts performance on the host side at all?

Reply:
Ran model perf and it seems ok, went red for one test but discussed with Austin and it's likely noise + being too close to the threshold even without my changes. Austin said he'll look at that test on main, and we'll revisit this block if it becomes an issue (hopefully we can keep this in, otherwise no way to know the issue Q write ptr after a run)

Reply:
Sounds good. In case its an actual perf regression, would you consider adding such functionality under debug build (I'm aware that release vs debug can mess with timing significantly, and make hang repro harder than it needs to be).

Reply:
I think the vision with this dump tool is that eventually customers can run it in the event of a hang and send out the relevant info, so my concern is that locking some of the info behind debug build would make that process difficult. If it does end up being a perf problem then maybe we just take it out and/or re-visit if we change hugepages

Reply:
Sounds good